### PR TITLE
feat(tools): calculate_distance_band — min/avg/max distance for N neighbors (ArcGIS Calculate Distance Band from Neighbor Count)

### LINKED ISSUE #415: feat(tools): calculate_distance_band — min/avg/max distance guaranteeing N neighbors per feature (ArcGIS Calculate Distance Band from Neighbor Count)
## Summary

Report the minimum / average / maximum distance required so that every feature has at least a given number of neighbors — the diagnostic used to pick a distance-band threshold before running distance-based spatial statistics.

**ArcGIS counterpart:** [Calculate Distance Band from Neighbor Count](https://pro.arcgis.com/en/pro-app/latest/tool-reference/spatial-statistics/calculate-distance-band-from-neighbor-count.htm)

## Motivation

Distance-band spatial stats need a threshold that guarantees no feature is left with zero neighbors. `generate_spatial_weights_matrix` will build a distance-band matrix but does not report the min/avg/max k-th-neighbor distance, so users currently guess the band. This is the standard companion diagnostic.

## Proposed behavior

- **Inputs:** `input` (point/feature layer); `neighbors` (k); `distance_method` (`euclidean` | `manhattan`).
- **Output:** the three summary distances (min, average, max of each feature's distance to its k-th nearest neighbor) — printed and/or written to a small table.

## Implementation sketch

For each feature, a `kdtree` k-nearest query returns the k-th neighbor distance (Euclidean or Manhattan); reduce to min/avg/max across all features. ~1-file tool reusing the vendored `kdtree`. 3-file pattern.

**Estimated effort:** easy

### PR DESCRIPTION
**Pull request by giswqs**
## Summary

Adds `calculate_distance_band` — reports the minimum, average, and maximum distance at which **every feature has at least `neighbors` neighbours**. Using the maximum as a distance band guarantees no feature is left with fewer than N neighbours.

**ArcGIS counterpart:** [Calculate Distance Band from Neighbor Count](https://pro.arcgis.com/en/pro-app/latest/tool-reference/spatial-statistics/calculate-distance-band-from-neighbor-count.htm)

This is the threshold diagnostic to run **before** any distance-based spatial statistic — `getis_ord_general_g`, incremental Moran's *I*, or a `fixed_distance_band` weights matrix — so no feature ends up with zero neighbours. `generate_spatial_weights_matrix` builds the matrix but never reports this band, so users currently guess the threshold.

## How it works

- For each feature, the vendored `kdtree` finds the distance to its `neighbors`-th nearest neighbour (`euclidean`, default, or `manhattan`).
- The query asks for `neighbors + 1` and drops the feature's **own index**, so a *different* feature at distance 0 (a coincident point) still counts as a neighbour.
- Returns `min_distance`, `avg_distance`, `max_distance`, plus an optional CSV of the per-feature k-th-neighbour distance.

Distances are in the layer's CRS units — use a projected CRS for metric bands.

## Implementation

3-file pattern: new `crates/geolibre-tools/src/calculate_distance_band.rs`, three wiring lines in `lib.rs`, one README row. Reuses the same `kdtree` KNN machinery as `generate_spatial_weights_matrix`. No new dependencies.

## Testing

**Unit tests** (5, all passing): unit-spaced chain 1st-neighbour band (min=avg=max=1), 2nd-neighbour band on the same chain (min=1 interior, max=2 endpoints), Manhattan vs Euclidean metric (L1=7 vs L2=5 on a 3-4 leg), coincident-point zero distance, and parameter rejection. Full crate suite: 1065 passing; clippy clean.

**Real-world validation.** Ran on 109 real US city points (`us_cities.geojson`) at k = 1, 3, 8 and compared against scikit-learn `NearestNeighbors`:

| k | tool (min / avg / max) | sklearn (min / avg / max) |
|---|------------------------|---------------------------|
| 1 | 0.117310 / 2.064027 / 8.180374 | 0.117310 / 2.064027 / 8.180374 |
| 3 | 1.431447 / 4.526104 / 37.284878 | 1.431447 / 4.526104 / 37.284878 |
| 8 | 3.846659 / 7.643084 / 40.277441 | 3.846659 / 7.643084 / 40.277441 |

Identical to 6 decimals. This tool emits a numeric diagnostic + CSV (not a spatial layer), so numeric agreement is the appropriate proof rather than a render.

Sample data under `../data`: `us_cities.geojson` (input), `calculate_distance_band_us_cities_k8.csv`.

Closes #415

## Summary by CodeRabbit

* **New Features**
  * Added the `calculate_distance_band` tool to compute per-feature *k*-th nearest-neighbor distance diagnostics (minimum/average/maximum), using either Euclidean or Manhattan distance.
  * Optionally exports the per-feature *k*-th neighbor distances to CSV.
  * Reports skipped non-point features and validates inputs (neighbor count, distance method, and sufficient point features).
* **Documentation**
  * Documented the new tool’s behavior and intended use for spatial-analysis pre-checks.
* **Tests**
  * Added coverage for Euclidean/Manhattan behavior, coincident-point handling, and invalid parameters.

### CODE REVIEW

File: crates/geolibre-tools/src/calculate_distance_band.rs [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+        // Representative point per feature; remember which feature it came from.
+        let mut pts: Vec<[f64; 2]> = Vec::new();
+        let mut feat_of: Vec<usize> = Vec::new();
+        for (fi, feature) in layer.features.iter().enumerate() {
+            if let Some((x, y)) = feature.geometry.as_ref().and_then(point_xy) {
+                pts.push([x, y]);
+                feat_of.push(fi);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (giswqs) [reply crates/geolibre-tools/src/calculate_distance_band.rs:110]: Done in 7b3de39 — non-point/multipoint features are now counted and surfaced (a progress log plus a `skipped_non_point` field in the outputs) so the band isn't silently computed over a subset.

File: crates/geolibre-tools/src/calculate_distance_band.rs [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+            .trim()
+            .parse::<usize>()
+            .map_err(|_| ToolError::Validation("'neighbors' must be an integer".to_string()))?
+            .max(1),
+        Some(_) => {
+            return Err(ToolError::Validation(
+                "'neighbors' must be a number".to_string(),
+            ))
+        }
+    };
```
[the remark this answers is not in the document]
Comment (giswqs) [reply crates/geolibre-tools/src/calculate_distance_band.rs:277]: Done in 7b3de39 — `neighbors` now rejects non-integers (3.5), zero, and negatives with a Validation error instead of coercing to 1; whole-valued floats like 3.0 are still accepted. Covered by rejects_bad_params.
